Makefile

```diff
@@ -0,0 +1,20 @@
+CPP     = g++
+CPPOPTS = -g -O3 -Wall -std=c++11 -Iinclude/ -Ideps/ -fprofile-arcs -ftest-coverage
```

Comment:
My understanding is that (1) you should not be using the `gcov` flags with the optimization flag (`O3`) to evaluate coverage and (2) building production code with the `gcov` flags will slow down the executable.  You might want to set the `CPPOPTS` according to the build type.